docs/bitrate-switching.md

```diff
@@ -0,0 +1,40 @@
+# Adaptive Switching Behavior
+The HLS tech tries to ensure the highest-quality viewing experience 
+possible, given the available bandwidth and encodings. This doesn't
+always mean using the highest-bitrate rendition available-- if the player
+is 300px by 150px, it would be a big waste of bandwidth to download a 4k
+stream. By default, the player attempts to load the highest-bitrate 
+variant that is less than the most recently detected segment bandwidth,
+with one condition: if there are multiple variants with dimensions greater
+than the current player size, it will only switch up one size greater 
+than the current player size.
+
+If you're the visual type, the whole process is illustrated
+below. Whenever a new segment is downloaded, we calculate the download
+bitrate based on the size of the segment and the time it took to
+download:
+
+![New bitrate info is available](bitrate-switching-1.png)
+
+First, we filter out all the renditions that have a higher bitrate
+than the new measurement:
```

Comment:
We should never discard the lowest available rendition based on bitrate. Otherwise we're left with no renditions to play at all. The question is: which is better, playing a rendition that will buffer due to bandwidth limits, or not play at all? In my opinion it is better to at least try to play because bandwith limits could be temporary or they could be caused by our bitrate sampling limitations.